### PR TITLE
KOGITO-7244: Patch PR for KOGITO-6806 with doc improvements in Saga example guide

### PR DESCRIPTION
**Pull request by hmanwani-rh**
**JIRA:** https://issues.redhat.com/browse/KOGITO-7244

**Description:** Patch PR for KOGITO-6806 with doc improvements in Saga example guide

https://github.com/kiegroup/kogito-docs/pull/37

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category

### CODE REVIEW

**Comment by hmanwani-rh** [comment]
@tiagodolphine Can you please review the changes that applied in this PR to the Saga example doc and share your feedback?

**Review by tiagodolphine — COMMENTED** [review]
Thanks, @hmanwani-rh LGTM, I just added a couple of comments.

File: serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc [hunk 1]
```diff
@@ -1,72 +1,64 @@
...
-
-While executing a step, failures might happen. In this situation, a sequence of compensating actions is executed to undo the changes made during the execution or leave the system in a known termination state, aiming for consistency.
+[[con-saga-overview]]
+== Overview of Saga pattern
 
-== The order fulfillment example
+The Saga design pattern manages data consistency across services that are available in distributed transaction scenarios. For more information about Saga pattern, see the initial link:https://www.cs.cornell.edu/andru/cs711/2002fa/reading/sagas.pdf[publication]. 
 
-This is a very common use case, present in many e-commerce companies and for sure it should be familiar to you.
+In a microservice architecture, you can define a service as microservice, which is responsible to perform actions related to the business domain.
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc:13]: Participant is used for a person, do you think we can use it to refer to microservices? If the audience is familiar with the term Participant then I am in for it. However, do you have any other term, which is less confusing?

File: serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc [hunk 2]
```diff
@@ -1,72 +1,64 @@
...
+.Example of order fulfillment process
+image::use-cases/orchestration-based-saga-pattern/order-fulfillment-example.png[Example of order fulfillment process]
 
-The below picture makes it possible to visualize the two main flows in case of success or failure during the order fulfillment flow, which describes a Saga pattern.
+The previous figure contains the following steps:
 
-.Example Order fulfillment Saga
-image::use-cases/orchestration-based-saga-pattern/order-fulfillment-saga.png[]
+1. *Order checkout*: User confirms the order with selected items, payment method, and delivery address.
+2. *Stock reservation*: The selected item in the order is removed from the reserved stock.
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc:30]: Done

File: serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc:13]: @tiagodolphine Thanks for sharing the reference. 
I've changed the microservice or service with the participant.

File: serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc [hunk 3]
```diff
@@ -99,20 +91,26 @@ The `transition` attribute represents the next step to be started, to perform th
...
 ----
 
-=== Compensations
+In the previous example, the `processPayment` state contains a `processPaymentAction` action, which invokes a function to execute the payment processing that `Payment Service` microservice might process. The `transition` attribute represents the next step to be started, which schedules the order shipping for the Saga pattern.
 
-The core functionality when designing the Saga is the compensation for each step to be executed by a participant.
+Compensation actions::
++
+--
+When designing a Saga pattern, compensation actions for each step is considered as a core functionality, which is executed by a microservice.
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc:99]: Done

**Comment by hmanwani-rh** [comment]
> Thanks, @hmanwani-rh LGTM, I just added a couple of comments.

@tiagodolphine Thanks for sharing the comments. I've made more changes as suggested. Please take a look one more time and let me know if this PR is good to merge.

**Review by tiagodolphine — APPROVED** [review]
Thanks @hmanwani-rh , LGTM

**Pull request merged**